Realm/RLMArray.mm

```diff
@@ -514,6 +514,13 @@ - (NSArray *)objectsAtIndexes:(NSIndexSet *)indexes {
     return [_backingArray objectsAtIndexes:indexes];
 }
 
+- (BOOL)isEqual:(id)object {
+    if (![object isKindOfClass:RLMArray.class]) {
+        return NO;
+    }
+    return [_backingArray isEqual:((RLMArray *)object)->_backingArray];
+}
```

Comment:
This'll return false for `[[RLMArray new] isEqual:[RLMArray new]]` because `_backingArray` is nil. I think the following is correct?

```suggestion
    if (auto array = RLMDynamicCast<RLMArray>(object)) {
        return !array.realm
            && ((_backingArray.count == 0 && array->_backingArray.count == 0)
                || [_backingArray isEqual:array->_backingArray]);
    }
    return NO;
```

The !array.realm check is needed as otherwise an empty unmanaged array will compare equal to all managed arrays. Checking that the count of both is 0 rather than checking for `nil` is because a nil backing array and an empty backing array are equal here.

Reply:
yep, good spot. Updated.